test/tools/ossfuzz/protoToAbiV2.cpp

```diff
@@ -724,14 +853,31 @@ void ProtoConverter::writeHelperFunctions()
 		return true;
 	}
 
-	/// Accepts function selector, correct argument encoding, and length of invalid encoding and returns
-	/// the correct and incorrect abi encoding for calling the function specified by the function selector.
-	function createEncoding(bytes4 funcSelector, bytes memory argumentEncoding, uint invalidLengthFuzz)
-		internal pure returns (bytes memory, bytes memory)
+	/// Accepts function selector, correct argument encoding, and length of
+	/// invalid encoding and returns the correct and incorrect abi encoding
+	/// for calling the function specified by the function selector.
+	function createEncoding(
+		bytes4 funcSelector,
+		bytes memory argumentEncoding,
+		uint invalidLengthFuzz,
+		bool isRightPadded
+	) internal pure returns (bytes memory, bytes memory)
 	{
 		bytes memory validEncoding = new bytes(4 + argumentEncoding.length);
-		// Ensure that invalidEncoding crops at least one and at most all bytes from correct encoding.
-		uint invalidLength = invalidLengthFuzz % argumentEncoding.length;
+		// Ensure that invalidEncoding crops at least 32 bytes (padding length
```

Comment:
Could this also be done in `ProtoConverter::visit(TestFunction)` such that the invalid length is passed directly? Or generally speaking, do we prefer to have as much code as possible on the C++ or on the Solidity side?

I also think that the `ProtoConverter` could be eventually split up such that Solidity "codegen", that's the string creation. But that's not in scope for this PR, I'd guess :)

Reply:
> Could this also be done in ProtoConverter::visit(TestFunction) such that the invalid length is passed directly?

Invalid length is first passed (in C++ code) here: https://github.com/ethereum/solidity/pull/7188/files#diff-7da1fb83eed279ed546eb9cb40618f9aR838

Since the C++ portion does not yet know what the size of the valid encoding is (we need to know the value of the valid encoding to place an upper bound on the length of the invalid encoding), we need to adjust the value passed from C++ inside solidity code.

Valid encoding size is computed in Solidity code by first making a call to abi encode on the parameter list here
https://github.com/ethereum/solidity/pull/7188/files#diff-7da1fb83eed279ed546eb9cb40618f9aR814

...and then adding the function selector size (4 bytes) here
https://github.com/ethereum/solidity/pull/7188/files#diff-7da1fb83eed279ed546eb9cb40618f9aR866

Reply:
> I also think that the ProtoConverter could be eventually split up such that Solidity "codegen", that's the string creation. But that's not in scope for this PR, I'd guess :)

Yeah, I will create an issue for this :-)

Reply:
@erak I was going to create an issue for this but realized I didn't follow your comment. Do you mind briefly commenting what you had in mind?